project-base/app/tests/FrontendApiBundle/Functional/Product/Flag/FlagsTest.php

```diff
@@ -19,34 +20,38 @@ public function testFlags(): void
             }
         ';
 
-        $jsonExpected = '{
-    "data": {
-        "flags": [
-            {
-                "name": "' . t('Action', [], Translator::DEFAULT_TRANSLATION_DOMAIN, $this->getFirstDomainLocale()) . '"
-            },
-            {
-                "name": "' . t('Made in CZ', [], Translator::DEFAULT_TRANSLATION_DOMAIN, $this->getFirstDomainLocale()) . '"
-            },
-            {
-                "name": "' . t('Made in DE', [], Translator::DEFAULT_TRANSLATION_DOMAIN, $this->getFirstDomainLocale()) . '"
-            },
-            {
-                "name": "' . t('Made in SK', [], Translator::DEFAULT_TRANSLATION_DOMAIN, $this->getFirstDomainLocale()) . '"
-            },
-            {
-                "name": "' . t('New', [], Translator::DEFAULT_TRANSLATION_DOMAIN, $this->getFirstDomainLocale()) . '"
-            },
-            {
-                "name": "' . t('Price hit', [], Translator::DEFAULT_TRANSLATION_DOMAIN, $this->getFirstDomainLocale()) . '"
-            },
-            {
-                "name": "' . t('Sale', [], Translator::DEFAULT_TRANSLATION_DOMAIN, $this->getFirstDomainLocale()) . '"
-            }
-        ]
-    }
-}';
+        $flags = [
+            [
+                'name' => t('Action', [], Translator::DEFAULT_TRANSLATION_DOMAIN, $this->getFirstDomainLocale()),
+            ],
+            [
+                'name' => t('Made in CZ', [], Translator::DEFAULT_TRANSLATION_DOMAIN, $this->getFirstDomainLocale()),
+            ],
+            [
+                'name' => t('Made in DE', [], Translator::DEFAULT_TRANSLATION_DOMAIN, $this->getFirstDomainLocale()),
+            ],
+            [
+                'name' => t('Made in SK', [], Translator::DEFAULT_TRANSLATION_DOMAIN, $this->getFirstDomainLocale()),
+            ],
+            [
+                'name' => t('New', [], Translator::DEFAULT_TRANSLATION_DOMAIN, $this->getFirstDomainLocale()),
+            ],
+            [
+                'name' => t('Price hit', [], Translator::DEFAULT_TRANSLATION_DOMAIN, $this->getFirstDomainLocale()),
+            ],
+            [
+                'name' => t('Sale', [], Translator::DEFAULT_TRANSLATION_DOMAIN, $this->getFirstDomainLocale()),
+            ],
+        ];
+
+        ArraySorter::sortArrayAlphabeticallyByValue('name', $flags);
+
+        $arrayExpected = [
+            'data' => [
+                'flags' => $flags,
+            ],
+        ];
 
-        $this->assertQueryWithExpectedJson($query, $jsonExpected);
+        $this->assertQueryWithExpectedJson($query, json_encode($arrayExpected, JSON_THROW_ON_ERROR));
```

Comment:
maybe `assertQueryWithExpectedArray` could be used instead of the encoding json, what do you think?

Reply:
Changed